### PR TITLE
Send a commit whenever we can create a PreparedCertificate

### PR DESCRIPTION
**Pull request by asaj**
### Description

This PR ensures we send a commit message whenever we create a PreparedCertificate. In particular, we were not sending a commit when we received a commit message that put us over 2F+1 prepares+commits.

### Tested

Not tested.

### Other changes

- Remove unused functions in `roundstate.go`
- Ensure we never build a `PreparedCertificate` larger than `quorumSize`

### Backwards compatibility

Backwards compatible

### CODE REVIEW

File: consensus/istanbul/core/roundstate.go [hunk 1]
```diff
@@ -141,51 +149,36 @@ func (s *roundState) Sequence() *big.Int {
...
-			Proposal:                s.Preprepare.Proposal,
-			PrepareOrCommitMessages: messages,
+		if s.Prepares.Get(message.Address) == nil {
+			messages[i] = *message
+			i++
 		}
-		return nil
-	} else {
+	}
+	if i != quorumSize {
```
Comment (nambrot) [inline consensus/istanbul/core/roundstate.go]: Question: Why do we need to `Ensure we never build a PreparedCertificate larger than quorumSize`?

File: consensus/istanbul/core/roundstate.go [hunk 2]
```diff
@@ -61,7 +61,15 @@ func (s *roundState) GetPrepareOrCommitSize() int {
 	s.mu.RLock()
 	defer s.mu.RUnlock()
 
-	return s.getPrepareOrCommitSize()
+	result := s.Prepares.Size() + s.Commits.Size()
```
[the remark this answers is not in the document]
Comment (asaj) [reply consensus/istanbul/core/roundstate.go]: It wasn't called anywhere else, so I just folded it into this function

File: consensus/istanbul/core/roundstate.go (hunk 1 above)
Comment (asaj) [reply consensus/istanbul/core/roundstate.go]: It's not strictly necessary

File: consensus/istanbul/core/commit.go [hunk 3]
```diff
@@ -111,11 +111,13 @@ func (c *core) handleCommit(msg *istanbul.Message) error {
...
 		c.commit()
 	} else if c.current.GetPrepareOrCommitSize() >= minQuorumSize && c.state.Cmp(StatePrepared) < 0 {
-		logger.Trace("Got enough prepares and commits to generate a PreparedCertificate")
 		if err := c.current.CreateAndSetPreparedCertificate(minQuorumSize); err != nil {
 			logger.Error("Failed to create and set preprared certificate", "err", err)
 			return err
 		}
+		logger.Trace("Got quorum prepares or commits", "tag", "stateTransition", "commits", c.current.Commits, "prepares", c.current.Prepares)
+		c.setState(StatePrepared)
+		c.sendCommit()
```
[the remark this answers is not in the document]
Comment (asaj) [reply consensus/istanbul/core/commit.go]: We could, is that something you'd prefer?

**Pull request merged**